content/00_Foreword.md

```diff
@@ -1,185 +1,105 @@
+
+The .NET runtime libraries have included the `IObservable<T>` and
+`IObserver<T>` interfaces that representing the core concept of reactive programming
+for well over a decade now. The Reactive Extensions for .NET are effectively a library of implementations of these
+interfaces. Rx.NET implementation first appeared
+back in 2010 but since then, Rx libraries have become available for other languages, and this way of programming has become
+especially popular in JavaScript.
+
+This book will introduce Rx via C#. The concepts are universal, so users of other .NET languages
+such as VB.NET and F#, will be able to extract the concepts and translate them to their particular
+language.
+
+Rx.NET is just a library, originally created by Microsoft, but now an open source project
+supported entirely through community effort. (Rx's current lead maintainer, [Ian Griffiths](https://endjin.com/who-we-are/our-people/ian-griffiths/),
+is also the author of the latest revision of this book, and indeed the author of this very
+sentence.)
+
+If you have never used Rx before, it _will_ change the way you design and build software.
+It provides a well thought out abstraction for a fundamentally important idea in computing—sequences
+of events. These are as important as lists or arrays, but before Rx there was little
+direct support in libraries or languages, and what support there was tended to be rather
+ad hoc, and built on weak theoretical underpinnings. Rx changes that. The extent to
+which this Microsoft invention has been wholehearted adopted by some developer communities
+traditionally not especially Microsoft-friendly is a testament to the quality of its
+fundamental design.
+
+This book aims to teach you:
+
+  * about the types that Rx defines
+  * about the extension methods Rx provides, and how to use them
+  * how to manage subscriptions to event sources
+  * how to visualize "sequences" of data and sketch your solution before coding it
+  * how to deal with concurrency to your advantage and avoid common pitfalls
+  * how to compose, aggregate and transform streams
+  * how to test your Rx code
+  * some common best practices when using Rx
+    
+The best way to learn Rx is to use it. Reading the theory from this book will only
+help you be familiar with Rx, but to fully understand it you should build things
+with it. So we warmly encourage you to build based on the examples in this book.
+
+
+# Acknowledgements    {#Acknowledgements}
+
+Firstly, I (Ian Griffiths) should make it clear that this revised edition builds
+on the excellent work of the original author Lee Campbell. I am grateful that he
+generously allowed the Rx.NET project to make use of his content, enabling this
+new edition to come into existence.
+
+I would also like to recognize the people that made this book
+possible. Crucial to the first edition of the book, in addition to the author, [Lee Campbell](https://leecampbell.com/), were: 
+[James Miles](http://enumeratethis.com/), 
```

Comment:
It might be worth checking these sites still exist - I know all the blogs.msdn.com have gone - maybe we should point to LinkedIn profiles instead, especially as many people have left Microsoft since the 1st edition.

I also wonder if we should point to the existance of A Little History of Reaqtor is people want a deeper insights into Rx and the teams and research projects involved.

Reply:
http://enumeratethis.com/ no longer exists